cluster-controller/src/main/java/io/strimzi/controller/cluster/operations/cluster/AbstractClusterOperations.java

```diff
@@ -253,6 +253,7 @@ public final void reconcile(String namespace, String name) {
                                 } else {
                                     log.error("Failed to update {} cluster {}.", clusterDescription, nameFromCm);
                                 }
+                                lock.release();
```

Comment:
Other than the different log messages, it seems that the handlers for called update/create/delete are the same. In your opinion does it make sense to factor out a common method ?

Reply:
I think the log message containing the operation is quite important (the handler might be called significant amount of time after the `reconcile` method, so the link to the other log messages describing what should be done will not be easy to establish). And IMHO if we write common handler which supports different log messages per operation, we might loose some simplicity and code readability. So I would say this is OK.

Reply:
What I see is that we have always common information which are `clusterDescription` and `nameFromCm` plus part of the log message which depends on the operation called (i.e. "updated", "update", "added", "add", "deleted", "delete"). Making a couple of generic phrases for success and failure log messages and then injecting the "operation" in the text should be one solution without adding complexity. Btw I'm fine if you want to leave it as it is now.